### PR TITLE
d8 implementation of the d7 version

### PR DESCRIPTION
**Pull request by mloyat**

### CODE REVIEW

**Comment by mloyat** [comment]
I made few changes in appbanners_include_tags to check the pages matches

File: appbanners.links.menu.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+appbanners.settings:
+  title: 'App Banners'
+  parent: system.admin_config
+  route_name: appbanners.admin_settings
+
```
Comment (malcomio) [inline appbanners.links.menu.yml]: the file shouldn't have trailing whitespace - please could you run this through PHP Code Sniffer if you can?

see https://www.drupal.org/node/1419980 and https://www.drupal.org/docs/develop/standards/coding-standards

File: src/Controller/AppbannersController.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\appbanners\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Component\Utility\Html;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+/**
+ * Provides route responses for the Example module.
```
Comment (malcomio) [inline src/Controller/AppbannersController.php]: copy and paste in the comment 😄

File: src/Controller/AppbannersController.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ */
+class AppbannersController extends ControllerBase {
+
+  /**
+   * Page callback for manifest.json.
+   */
+  public function manifestPage() {
+
+    $config = \Drupal::config('appbanners.settings');
```
Comment (malcomio) [inline src/Controller/AppbannersController.php]: ideally we'd use dependency injection for some of this stuff, but we can worry about that later - more valuable to get a working D8 version first

File: src/Controller/AppbannersController.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      'prefer_related_applications' => TRUE,
+      'related_applications' => [
+        [
+          'platform' => 'play',
+          'id' => Html::escape($config->get('android_app_id')),
+        ],
+      ],
+    ];
+
+    if (NULL != $name = $config->get('android_name')) {
```
Comment (malcomio) [inline src/Controller/AppbannersController.php]: generally a good idea to avoid assignment in conditions

Comment (malcomio) [reply src/Controller/AppbannersController.php]: I need to check coding standards, but I'd suggest doing this as 

```
$name = $config->get('android_name');
if (!empty($name)) {
```

File: src/Form/SettingsForm.php [hunk 5]
```diff
@@ -0,0 +1,160 @@
+<?php
```
Comment (malcomio) [inline src/Form/SettingsForm.php:1]: we should add a schema.yml for the settings: https://www.drupal.org/docs/8/api/configuration-api/configuration-schemametadata

File: src/Controller/AppbannersController.php (hunk 3 above)
Comment (mloyat) [reply src/Controller/AppbannersController.php]: wow, this one was strong for a lonely newbie ^^'

**Review by mloyat — COMMENTED** [review]
Hi @malcomio,
thanks for your comments, learning in progress ;) 
I updated my code, hope i did it right.

**Comment by mloyat** [comment]
Added config schema.
Added config object injection in controller.

File: appbanners.info.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+name: App Banners
+description: Provides support for Apple Smart App Banners and Android native app install banners.
+type: module
+core: 8.x
+
+configure: appbanners.admin_settings
```
Comment (malcomio) [inline appbanners.info.yml:6]: > All text files should end in a single newline (\n).
https://www.drupal.org/docs/develop/standards/coding-standards

```suggestion
configure: appbanners.admin_settings

```

File: appbanners.links.menu.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+appbanners.settings:
+  title: 'App Banners'
+  parent: system.admin_config
+  route_name: appbanners.admin_settings
```
Comment (malcomio) [inline appbanners.links.menu.yml:4]:
```suggestion
  route_name: appbanners.admin_settings

```

File: appbanners.permissions.yml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+administer app banners:
+  title: 'Administer App Banners'
```
Comment (malcomio) [inline appbanners.permissions.yml:2]:
```suggestion
  title: 'Administer App Banners'

```

File: appbanners.routing.yml [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  requirements:
+    _permission: 'administer app banners'
+
+appbanners.android_manifest:
+  path: '/appbanners/manifest.json'
+  defaults:
+    _controller: '\Drupal\appbanners\Controller\AppbannersController::manifestPage'
+    _title: 'Android manifest'
+  requirements:
+    _permission: 'access content'
```
Comment (malcomio) [inline appbanners.routing.yml:15]:
```suggestion
    _permission: 'access content'

```

File: config/install/appbanners.settings.yml [hunk 10]
```diff
@@ -0,0 +1,2 @@
+admin: FALSE
+visibility: exclude
```
Comment (malcomio) [inline config/install/appbanners.settings.yml:2]:
```suggestion
visibility: exclude

```

File: config/schema/appbanners.schema.yml [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+    admin:
+      type: boolean
+      label: 'Include app banners tags on admin pages?'
+    visibility:
+      type: string
+      label: 'Wether to nclude or exclude pages listed in "pages" field'
+    pages:
+      type: text
+      label: 'Include script on specific pages'
```
Comment (malcomio) [inline config/schema/appbanners.schema.yml:36]:
```suggestion
      label: 'Include script on specific pages'

```

**Comment by malcomio** [comment]
I've addressed these comments in #2

**Pull request closed without merging**